sentry-apollo-3/src/main/java/io/sentry/apollo3/SentryApollo3HttpInterceptor.kt

```diff
@@ -119,34 +115,44 @@ class SentryApollo3HttpInterceptor(private val hub: IHub = HubAdapter.getInstanc
         }
         newSpan.finish()
```

Comment:
`BeforeSpanCallback` may return `null`, why does this even compile?

Reply:
The BeforeSpanCallback is currently implemented as returning a NonNull Value. Same as in ApolloInterceptor V2. Returning null is dropping the span, correct? I think we should open an issue for V2 then.

Reply:
Ah oversight on my part, didn't see there's multiple interfaces with different signatures. Created https://github.com/getsentry/sentry-java/issues/2149 to changes this in v2 to allow returning `null`.